src/path_manager.c

```diff
@@ -105,86 +105,158 @@ static char const *get_pm_name(void const *data, size_t len)
 }
 #endif // MPTCPD_ENABLE_PM_NAME
 
-static void handle_connection_created(struct l_genl_msg *msg,
-                                      void *user_data)
+
+/**
+ * @struct pm_event_attrs
+ *
+ * @brief MPTCP generic netlink attribute values.
+ */
+struct pm_event_attrs
+{
+        /// MPTCP connection token.
+        mptcpd_token_t const *token;
+
+        /// Local MPTCP address ID.
+        mptcpd_aid_t const *laddr_id;
+
+        /// Remote MPTCP address ID.
+        mptcpd_aid_t const *raddr_id;
+
+        /// Local IPv4 address.
+        in_addr_t laddr4;
```

Comment:
Using the value here (and with `raddr4`) rather than a pointer makes it incompatible with MPTCP_GET_NL_ATTR (leading to the CI build break and the same on my machine).

Converting to a value rather than pointer also means 0.0.0.0 is treated as "no address" rather than "all local addresses", not sure if 0.0.0.0 is used in any events? Could be useful to allow it for future flexibility.

Reply:
> Using the value here (and with `raddr4`) rather than a pointer makes it incompatible with MPTCP_GET_NL_ATTR (leading to the CI build break and the same on my machine).

Right.  I was going to modify `MPTCP_GET_NL_ADDR` to return a value to address this build error but I'll hold off on that given your comment below.

> Converting to a value rather than pointer also means 0.0.0.0 is treated as "no address" rather than "all local addresses", not sure if 0.0.0.0 is used in any events? Could be useful to allow it for future flexibility.

That's fine.  I'll stick with a pointer to `in_addr_t` instead.